pkg/baseconn/conn.go

```diff
@@ -89,13 +94,18 @@ func (conn *BaseConn) QuerySQL(tctx *tcontext.Context, query string, args ...int
 	if conn == nil || conn.DB == nil {
 		return nil, terror.ErrDBUnExpect.Generate("database connection not valid")
 	}
-	tctx.L().Debug("query statement", zap.String("query", query), zap.Reflect("argument", args))
+	tctx.L().Debug("query statement",
+		zap.String("query", utils.TruncateString(query, stringLenLimit)),
+		zap.Reflect("argument", utils.TruncateInterface(args, stringLenLimit)))
 
 	rows, err := conn.DB.QueryContext(tctx.Context(), query, args...)
 
 	if err != nil {
-		tctx.L().Error("query statement failed", zap.String("query", query), zap.Reflect("argument", args), log.ShortError(err))
-		return nil, terror.ErrDBQueryFailed.Delegate(err, query)
+		tctx.L().Error("query statement failed",
+			zap.String("query", utils.TruncateString(query, stringLenLimit)),
+			zap.Reflect("argument", utils.TruncateInterface(args, stringLenLimit)),
+			log.ShortError(err))
+		return nil, terror.ErrDBQueryFailed.Delegate(err, utils.TruncateString(query, stringLenLimit))
```

Comment:
how about remain the whole error message, only do TruncateString when need print it

Reply:
the error message can be logged/sent by RPC in many places, it's hard to find out all of them, and ensure truncate it in later usage.

Reply:
any better suggestion?

Reply:
ok, no better suggestion